lib/Controller/LoginController.php

```diff
@@ -358,32 +358,40 @@ private function login($uid, Profile $profile, $newGroupPrefix = '')
                 $groupNames = $profile->data['groups'];
                 $groupMapping = isset($profile->data['group_mapping']) ? $profile->data['group_mapping'] : null;
                 $userGroups = $this->groupManager->getUserGroups($user);
+                $autoCreateGroups = $this->config->getAppValue($this->appName, 'auto_create_groups');
 
                 if ($groupMapping) {
+                    foreach ($userGroups as $group) {
+                        if (in_array($group->getGID(), array_values($groupMapping))) {
+                            $group->removeUser($user);
```

Comment:
What purpose of this ? Remove user from group and then add it to same group below ? :smile:

Reply:
Good point, I changed the logic back.